packages/i18n/src/create-i18n.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * External dependencies
+ */
+import Tannin from 'tannin';
+
+/**
+ * @typedef {{[key: string]: any}} LocaleData
+ */
+
+/**
+ * Default locale data to use for Tannin domain when not otherwise provided.
+ * Assumes an English plural forms expression.
+ *
+ * @type {LocaleData}
+ */
+const DEFAULT_LOCALE_DATA = {
+	'': {
+		plural_forms: ( n ) => ( n === 1 ? 0 : 1 ),
+	},
+};
+
+/**
+ * An i18n instance
+ * @typedef {Object} I18n
+ * @property {Function} setLocaleData Merges locale data into the Tannin instance by domain. Accepts data in a
+ *																		Jed-formatted JSON object shape.
+ * @property {Function} __ 						Retrieve the translation of text.
+ * @property {Function} _x 						Retrieve translated string with gettext context.
+ * @property {Function} _n 						Translates and retrieves the singular or plural form based on the supplied
+ * 																		number.
+ * @property {Function} _nx 					Translates and retrieves the singular or plural form based on the supplied
+ *																		number, with gettext context.
+ * @property {Function} isRTL 				Check if current locale is RTL.
+ */
+
+/**
+ * Create an i18n instance
+ *
+ * @param {LocaleData} [initialData]   	Locale data configuration.
+ * @param {string}     [initialDomain] 	Domain for which configuration applies.
+ * @return {I18n} 											I18n instance
```

Comment:
It looks like there's a mix of tabs + spaces for alignment of types/names/descriptions. I believe spaces should be used for consistency.